### ISSUE TITLE
mixed up kanji

### ISSUE THREAD

**Issue by dakkar** [mvysny/aedict#919]
AEdict 3.50.17 from Google Play, latest dictionaries as of 2020-05-27 (I deleted and re-downloaded to make sure), using the built-in font, on OnePlus 7pro, Android 10

Screenshot:
![image](https://user-images.githubusercontent.com/86328/83060246-0e3f9880-a053-11ea-8238-ef013f1636d1.png)

I got this by searching for "rei" (command, order).

Definitnely correct:
* kanji at the top, next to the menu 令
* meanings
* stroke order diagram

Definitely incorrect:
* large kanji 令
* parts

I'm not sure about the rest.

**Comment by dakkar** [comment]
update: apparently the two kanji *can be* the same kanji, depending on the font (?), and I really have no clue how I managed to get both representations in the text above.

(aedict's display is still a bit confusing, though!)

**Comment by topherbuckley** [comment]
Yep, this is a fairly common issue to see mismatches, but not specific to Aedict per se. It seems to be a non-issue on any phones I buy here in Japan with the default fonts, but with any international phones, these fonts tend to vary and are typically defaulted to render kanji using a Chinese font. In your case it looks like the opposite (i.e the font you thought was correct, was the incorrect one). See [here](https://en.m.wiktionary.org/wiki/%E4%BB%A4). 

Your system font would be displayed in any text search boxes (one at the top of your screenshot), the large font would be the embedded one within Aedict, and the stroke diagram would be from whichever database Aedict is currently pulling stroke diagrams from . Which database to pull from also seemed to be a topic of debate in past issues.

In conclusion, you have the power to change your system font to a true Japanese font, which will fix the top one. You can also change the font of the large Kanji via the Aedict settings. The stroke order diagram looks to be an issue with the stroke order database which is outside Aedicts control. Could be a better database somewhere though?

@mvysny what is the default packaged font and stroke order database used with Aedict currently? 

See a few past issues for further details:
https://github.com/mvysny/aedict/issues/682
https://github.com/mvysny/aedict/issues/490
https://github.com/mvysny/aedict/issues/469
https://github.com/mvysny/aedict/issues/211
https://github.com/mvysny/aedict/issues/113

**Comment by mvysny** [comment]
@topherbuckley thank you for an excellent summary!

Yes. The topmost kanji next to the menu button is drawn by Android system and Aedict has no way of controlling that one. You will need to modify your system settings in order to change that.

Regarding the "large kanji" view: Aedict bundles its own font called `DroidSansJapanese.ttf` which should hopefully contain correct glyphs for all Japanese characters. This font is then applied by default to the "large kanji" view, but this behaviour can be modified.

You can head to Settings / Expert Settings / Japanese Font to control those:

* `Do not use Japanese font`: will stop using DroidSansJapanese.ttf and will set the locale of the "large kanji" view to Japanese. This will start rendering Japanese glyphs correctly on some phones, but will do nothing on others and will still use their Chinese counterparts.
* `Custom Japanese Font for large characters`: you can supply your own TTF font file which will then be loaded by Aedict and used to draw the large characters. You need to download the font from somewhere to your phone first. The issue #490 mentions a couple of excellent tips. In order for this to work, `Do not use Japanese font` must be unchecked. (sorry about the double negative)

**Comment by mvysny** [comment]
[DroidSansJapanese.ttf.zip](https://github.com/mvysny/aedict/files/4693459/DroidSansJapanese.ttf.zip)

This is the font which Aedict bundles in. Perhaps it could be verified whether the glyphs in this font are correct and perhaps a better font can be bundled instead? However, I think I downloaded this font from official Google Android page so it should be in a good shape (even though it could be a couple of years old).

**Comment by mvysny** [comment]
Regarding the parts: unfortunately some of the parts do not have any representation in any font! For example the  `^` part - the character always draws with the `|` part as well. Perhaps I should simply draw the caret `^` instead, to avoid confusion...
